### PR TITLE
[PT-264] Add build init step and tests

### PR DESCRIPTION
**Pull request by NaurisSadovskis**
# Main changes:
- Add `delivery-helper init build` step that fetches git info and prints release variables to be used in the pipeline
- Add tests for `build/`, `slackNotify/` and `helper/`
- Add some dependencies (Go Git)
- Update README.md and bump version

# Usage:
For build variables to be applied, you must wrap delivery-helper command inside `export`, like this:
```
$ export $(delivery-helper init build)
```
This is due to Go not being able to set env vars for parent process.

### CODE REVIEW

File: build/github_environment.go [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+	"strings"
+
+	"gopkg.in/src-d/go-git.v4"
+)
+
+func getGitEnvironment() (map[string]string, error) {
+	gitDetails := make(map[string]string)
+
+	gitDetails, err := readGitRepository(gitDetails)
+	if err != nil {
```
Comment (darkraiden) [inline build/github_environment.go]: This if can be removed. If `readGitRepository` doesn't fail, it'll return `nil`, ergo `err === nil`

The function itself expects you to return two values either way and will be checked once it'll be run.

File: build/github_environment.go [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+		return gitDetails, err
+	}
+
+	return gitDetails, nil
```
Comment (darkraiden) [inline build/github_environment.go]: As a result of my previous comment, `return gitDetails, err`?

File: build/github_environment.go [hunk 3]
```diff
@@ -0,0 +1,76 @@
...
+	result := r.FindStringSubmatch(remote)
+
+	return result, nil
+}
+
+func readGitRepository(g map[string]string) (map[string]string, error) {
+	// Open repository from the current directory
+	repo, err := git.PlainOpen("")
+	if err != nil {
+		return g, err
```
Comment (darkraiden) [inline build/github_environment.go]: If function fails, no need to return `g`, instead `nil` can be returned

File: build/github_environment.go [hunk 4]
```diff
@@ -0,0 +1,76 @@
...
+	}
+
+	head, err := repo.Head()
+	if err != nil {
```
Comment (darkraiden) [inline build/github_environment.go]: same as above

File: build/github_environment.go [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+		return g, err
+	}
+
+	remote, err := repo.Remote(git.DefaultRemoteName)
+	if err != nil {
```
Comment (darkraiden) [inline build/github_environment.go]: same as above

File: build/github_environment.go [hunk 6]
```diff
@@ -0,0 +1,76 @@
...
+		return g, err
+	}
+
+	// Grab the first remote URL
+	remoteURL := remote.Config().URLs[0]
+
+	// Extract Git Organisation & Git Repo from the URL
+	remoteInfo, err := parseRemote(remoteURL)
+	if err != nil {
```
Comment (darkraiden) [inline build/github_environment.go]: same as above

Comment (NaurisSadovskis) [reply build/github_environment.go]: Done 👍

**Pull request merged**